variables.tf

```
@@ -0,0 +1,29 @@
+variable "log_analytics_workspace" {
+  type = object({
+    id                  = string
+    name                = string
+    resource_group_name = string
+    location            = string
+  })
+  description = "Log Analytics Worksapce that all VMs are connected to for monitoring"
+}
+
+variable "webhook_name" {
+  type = string
+  description = "Name of the alert webhook"
+}
+
+variable "webhook_service_uri" {
+  type = string
+  description = "Link to the webhook receiver URL"
+}
+
+variable "additional_queries" {
+  type = map(object({
+    query_path  = string
+    description = string
+    time_window = number
+  }))
+  description = "List of additional alert rule queries to create with a file path, description and time_window"
+  default = {}
+}
```

```suggestion
variable "log_analytics_workspace" {
  type = object({
    id                  = string
    name                = string
    resource_group_name = string
    location            = string
  })
  description = "Log Analytics Worksapce that all VMs are connected to for monitoring"
  nullable = false
}

variable "webhook_name" {
  type = string
  description = "Name of the alert webhook"
  nullable = false
}

variable "webhook_service_uri" {
  type = string
  description = "Link to the webhook receiver URL"
  nullable = false
}

variable "additional_queries" {
  type = map(object({
    query_path  = string
    description = string
    time_window = number
  }))
  description = "List of additional alert rule queries to create with a file path, description and time_window"
  default = {}
  nullable = false
}
```
I suppose you don't want `null` to be a valid input. See [Terraform Documentation on Input variables](https://developer.hashicorp.com/terraform/language/values/variables#disallowing-null-input-values). This forces you to require terraform version `1.1.0` but I think that's not that bad anyway :D.